webapp/channels/src/packages/mattermost-redux/src/reducers/entities/scheduled_posts.ts

```diff
@@ -82,39 +106,59 @@ function errorsByTeamId(state: ScheduledPostsState['errorsByTeamId'] = {}, actio
         Object.keys(scheduledPostsByTeamId).forEach((teamId: string) => {
             if (scheduledPostsByTeamId.hasOwnProperty(teamId)) {
                 const teamScheduledPosts = scheduledPostsByTeamId[teamId] as ScheduledPost[];
-                const updatedHasError = teamScheduledPosts.some((scheduledPost) => scheduledPost.error_code);
+                const scheduledPostIDsWithError = teamScheduledPosts.
+                    filter((scheduledPost) => scheduledPost.error_code).
+                    map((scheduledPost) => scheduledPost.id);
 
-                if (state[teamId] !== updatedHasError) {
+                if (state[teamId] !== scheduledPostIDsWithError) {
                     changed = true;
-                    newState[teamId] = updatedHasError;
+                    newState[teamId] = scheduledPostIDsWithError;
                 }
             }
         });
 
         return changed ? newState : state;
     }
     case ScheduledPostTypes.SINGLE_SCHEDULED_POST_RECEIVED: {
+        let changed = false;
+
         const teamId = action.data.teamId || 'directChannels';
+        const newState = {...state};
+        if (!newState[teamId]) {
+            newState[teamId] = [];
+        }
 
-        // if team already has an error state, then irrespective of what's the error
-        // on new scheduled post, the team would still have an error.
-        // So nothing changes so we return the original state as-in.
-        if (state[teamId]) {
-            return state;
+        const scheduledPost = action.data.scheduledPost as ScheduledPost;
+        if (scheduledPost.error_code) {
+            const alreadyExists = newState[teamId].find((scheduledPostId) => scheduledPostId === scheduledPost.id);
+            if (!alreadyExists) {
+                newState[teamId] = [...newState[teamId], scheduledPost.id];
+                changed = true;
+            }
         }
 
+        return changed ? newState : state;
+    }
+    case ScheduledPostTypes.SCHEDULED_POST_DELETED: {
+        let changed = false;
+
         const scheduledPost = action.data.scheduledPost as ScheduledPost;
+        const newState = {...state};
 
-        // if team doesn't have any error and neither does the new scheduled post,
-        // then nothing changes so we return the original state as-in.
-        if (!scheduledPost.error_code || state[teamId]) {
-            return state;
-        }
+        Object.keys(state).some((teamId: string) => {
```

Comment:
Same comment here as above about using a for..in here

Reply:
Done.